plugins/kubernetes/app/models/kuber_deploy_service.rb

```diff
@@ -44,14 +44,49 @@ def watch_deployment
     # TODO: hunt down this issue and fix it properly
     Watchers::PodWatcher ; Watchers::ReplicationControllerWatcher ; Watchers::EventWatcher
 
-    # Commented out for now, to avoid creating threads without ever destroying them
-    # Thread.new { kuber_release.watch_pods }
-    # Thread.new { kuber_release.watch_rcs }
+    Thread.new do
+      kuber_release.watch_pods(&method(:handle_pod_notice))
+    end
+
     # Thread.new { sleep(5) ; kuber_release.watch_pod_events }
   end
 
   private
 
+  def handle_pod_notice(notice, watcher)
+    release_doc = kuber_release.docs_by_role[notice.role]
+
+    if release_doc.nil?
+      Kubernetes::Util.log "ERROR: could not find role", role: notice.role, available_roles: kuber_release.docs_by_role.keys
+      return
+    end
+
+    ready_count = watcher.num_ready
+    if ready_count > release_doc.replicas_live
+      release_doc.update_replica_count(ready_count)
+      release_doc.save!
+      update_release_status
+      log 'Another replica is live', role: notice.role, count: ready_count, r_status: kuber_release.status, r_doc_status: release_doc.status
+    end
+  end
+
+  def update_release_status
+    statuses = kuber_release.release_docs.map(&:status)
+
+    if statuses.all? { |s| s == 'live' }
```

Comment:
The set of valid values for status are defined in `Kubernetes::Release::STATUSES`.  Ruby doesn't have a built-in Enum construct in the way that some other languages do.  And in the database, the `status` field is stored as a string.

So I have a few options:
1. Define constants, so it would look like `statuses.all? { |s| s == Kubernetes::ReleaseDoc::STATUS_LIVE }`
2. Create a method for each status on the object itself, e.g. `statuses.all?(&:live)`
3. Override the `#status` method of ReleaseDoc so it returns a symbol instead of a string, and the `#status=` method so it accepts either a symbol or a string.

All 3 options seem like extra work for not a whole lot of gain, but I'm willing to be convinced otherwise.